aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -229,7 +229,10 @@ Tensor lu_det_P(const Tensor& pivots) {
 TORCH_IMPL_FUNC(_linalg_det_out)(const Tensor& A, const Tensor& result, const Tensor& LU, const Tensor& pivots) {
   // info is an aux tensor
   auto info = at::empty({0}, A.options().dtype(kInt));
-  at::linalg_lu_factor_ex_out(const_cast<Tensor&>(LU), const_cast<Tensor&>(pivots), const_cast<Tensor&>(info), A);
+  // Optimisation: lu_factor_ex requires the input to be F-contig, otherwise it copies
+  // Use the transpose of if A is contiguous since det(A^T) = det(A)
+  // We limit this to real matrices, but it could also be implemented for complex matrices
```

Comment:
We does it not work for complex inputs?

Reply:
We are saving the LU decomposition for the backward. In the backward I use `linalg.lu_solve`, which just has the `adjoint` flag. It could be done by exposing the `transpose` flag somehow but yeah.
Another way of going about it would be by marking this function as a fallthrough for complex conj, and checking that bit, but yeah, for the next time.

Reply:
Note that this is an optimisation that then we need to take into account in the backward. This same optimisation is done in `linalg_solve`.